newsAnalysis/datastore/collect.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+'''
+Handles transport of big data for analysis
+'''
+from datastore.model import Article
+import unicodecsv as csv
+import os
+
+
+def download_articles_to_file(session, file_path, source_id=1, limit=None):
+    '''
+    session
+        An SqlAlchemy Session
+    file_path : String
+        The path of the file to write to. Will be appended to if it exists and
+        already contains data.
+    source_id : integer
+        The id of the source. For example: The source_id of Moscow Times is 1.
+    limit : Integer
+        The maximum amount of articles to be downloaded
+    '''
+
+    # Find the last id to resume with
+    print ("detecting latest id")
+    downloaded_articles_count = 0
+    last_id = 0
+    file_exists = os.path.isfile(file_path)
+    if file_exists:
+        with open(file_path, "rb") as f:
+            reader = csv.reader(f)
+            for row in reader:
+                last_id = int(row[0])
+                downloaded_articles_count += 1
+    print ("%d aricles have already been downloaded" % downloaded_articles_count)
+    print ("found latest id %d" % last_id)
+
+    corrected_limit = None
+    if limit is not None:
+        corrected_limit = max(0, limit - downloaded_articles_count)
+    print("corrected_limit = %r" % corrected_limit)
+
+    if corrected_limit is None or corrected_limit > 0:
+        query = session.query(Article)\
```

Comment:
Limiting the number of documents and adding new ones to the csv file work like a charm!
However, I some, ~10%, empty rows (only id and source_id, but no published_date, title or body) in the csv.
Maybe an issue with query and its filter?

Reply:
Do you have any example ids?

Reply:
I don't think it's the filter as less then 1% of rows has incomplete data. That would be a pretty big coincidence. Perhaps something goes wrong when writing or reading (special chars maybe).

I will write a test for those ids and look into it. If csv is unreliable I'll switch to binary

Reply:
I get 9 empty rows out of 100. Sample ids are 3,9,16,22

Reply:
Maybe an encoding issue with the csv writer? But then it's strange that there is no date shown...